sycl/include/CL/sycl/detail/image_accessor_util.hpp

```diff
@@ -855,6 +860,88 @@ DataTy ReadPixelDataNearestFiltMode(cl_int4 PixelCoord, id<3> ImgPitch,
   return Color;
 }
 
+// Checks if the PixelCoord is out-of-range, and returns appropriate border or
+// color value at the PixelCoord.
+template <typename DataT>
+DataT getColor(cl_int4 PixelCoord, addressing_mode SmplAddrMode,
+               range<3> ImgRange, id<3> ImgPitch,
+               image_channel_type ImgChannelType,
+               image_channel_order ImgChannelOrder, void *BasePtr,
+               uint8_t ElementSize) {
+  DataT RetData;
+  if (isOutOfRange(PixelCoord, SmplAddrMode, ImgRange)) {
+    cl_float4 BorderColor = (getBorderColor(ImgChannelOrder));
+    RetData = BorderColor.convert<typename TryToGetElementType<DataT>::type>();
+  } else {
+    RetData = ReadPixelData<DataT>(PixelCoord, ImgPitch, ImgChannelType,
+                                   ImgChannelOrder, BasePtr, ElementSize);
+  }
+  return RetData;
+}
+
+// Computes and returns color value with Linear Filter Mode.
+// Steps:
+// 1. Computes the 8 coordinates using all combinations of i0/i1,j0/j1,k0/k1.
+// 2. Calls getColor() on each Coordinate.(Ci*j*k*)
+// 3. Computes the return Color Value using a,b,c and the Color values. 
+template <typename DataT>
+DataT ReadPixelDataLinearFiltMode(cl_int8 CoordValues, cl_float4 abc,
+                                  addressing_mode SmplAddrMode,
+                                  range<3> ImgRange, id<3> ImgPitch,
+                                  image_channel_type ImgChannelType,
+                                  image_channel_order ImgChannelOrder,
+                                  void *BasePtr, uint8_t ElementSize) {
+  cl_int i0 = CoordValues.s0(), j0 = CoordValues.s1(), k0 = CoordValues.s2(),
```

Comment:
why are you using a different style here?

Reply:
Different style?
You mean why am i extracting the elements out? By storing them in i0,j0, it's easier to read the code and find mistakes.
The computations are still done on the whole vector, when computing the RetData value.

Reply:
oh , or do you mean using s0? Because .x/.y/.z work for elements lesser than or equal to 4.

Reply:
I mean this difference:
`int a,b,c;`
and 
```
int a;
int b;
int c;
```

Reply:
Let me know what you meant here.

Reply:
Just a different state of mind I think.
Do you want me to do that change?

Reply:
I just asked why)